Makefile

```diff
@@ -1,12 +1,21 @@
 #
-# github.com/docker/cli 
+# github.com/docker/cli
 #
 
+VERSION ?= unknown-version
+GITCOMMIT ?= $(shell git rev-parse --short HEAD 2> /dev/null || true)
+BUILDTIME ?= $(shell date --utc --rfc-3339 ns 2> /dev/null | sed -e 's/ /T/')
+
+LDFLAGS := -X github.com/docker/cli/cli.GitCommit=${GITCOMMIT} \
+	-X github.com/docker/cli/cli.BuildTime=${BUILDTIME} \
+	-X github.com/docker/cli/cli.Version=${VERSION} \
```

Comment:
It's possible I'm misunderstanding what you want to achieve, but...

I think you can do
```
target: FOO := BAR
```
to only set `$(FOO)` for the `targetA` and `targetB` (I'm not 100% sure this isn't a gnu-make-ism though), whether you do that in the main `Makefile` or via an `include` should be independent.